### PR TITLE
Add new animations

### PR DESCRIPTION
**Pull request by lilvedaes**
- Side movement animation
 - Celebration animation
 - Falling to death animation

### CODE REVIEW

File: Assets/Scripts/ObjectScripts/Checkpoint.cs [hunk 1]
```diff
@@ -4,6 +4,8 @@
 public class Checkpoint : MonoBehaviour
```
Comment (jamilfaisal) [inline Assets/Scripts/ObjectScripts/Checkpoint.cs:4]: Are any of the changes in this file needed? It seems like the animation is commented out on line 36.

**Review by jamilfaisal — CHANGES_REQUESTED** [review]
1. I don't see the side movement animation happening when I switch lanes. It's supposed to be a small shuffle, but I don't see that. Is the running animation overriding the side movement animation?
2. For the cat falling animation when you die, the cat disappears for a bit and then reappears in the middle of the animation. Is this intentional?
3. You can't really see the winning animation because the win screen pops up as soon as you enter the endbox. I think we can delay the win screen. Also, it would be cool to make the camera face in front of the cat (instead of behind), hide the endbox, and play the win animation, and then show the win screen. I'm imagining something similar to when you beat a level in sonic adventure DX: https://youtu.be/fzGn-OEW2oM?t=1891

File: Assets/Scripts/ObjectScripts/Checkpoint.cs (hunk 1 above)
Comment (lilvedaes) [reply Assets/Scripts/ObjectScripts/Checkpoint.cs:4]: yeah I wanted to leave the commented-out line in there for when we get the checkpoint animation revised edition so it's easier to add it in

**Comment by lilvedaes** [comment]
> 1. I don't see the side movement animation happening when I switch lanes. It's supposed to be a small shuffle, but I don't see that. Is the running animation overriding the side movement animation?
> 2. For the cat falling animation when you die, the cat disappears for a bit and then reappears in the middle of the animation. Is this intentional?
> 3. You can't really see the winning animation because the win screen pops up as soon as you enter the endbox. I think we can delay the win screen. Also, it would be cool to make the camera face in front of the cat (instead of behind), hide the endbox, and play the win animation, and then show the win screen. I'm imagining something similar to when you beat a level in sonic adventure DX: https://youtu.be/fzGn-OEW2oM?t=1891

1 -> It is playing, it's just very small since the side movement is just a tiny jump that is not too visible. I tried making it also move up when this animation is played so it's more noticeable but it looks weird and disturbs the other animations. Maybe we could brainstorm new side movement animations that are more noticeable on Saturday with the art team?
2 -> Yeaa there's a blink effect
3 -> Oh yeah I can look into that, that's a cool idea, but since the cat goes inside the box we would not really see their face. I could try having the cat stop before we get to the box and then continue moving forward, I don't think any of my other tasks have more priority than this one besides the playtesting report.

File: Assets/Scripts/ObjectScripts/Checkpoint.cs (hunk 1 above)
Comment (jamilfaisal) [reply Assets/Scripts/ObjectScripts/Checkpoint.cs:4]: I see, that makes sense! Can you add a quick comment above line 36 like "// TODO: Add revised checkpoint animation"? Just so the other devs know why it's commented out. Thanks 😃

**Comment by jamilfaisal** [comment]
> > 1. I don't see the side movement animation happening when I switch lanes. It's supposed to be a small shuffle, but I don't see that. Is the running animation overriding the side movement animation?
> > 2. For the cat falling animation when you die, the cat disappears for a bit and then reappears in the middle of the animation. Is this intentional?
> > 3. You can't really see the winning animation because the win screen pops up as soon as you enter the endbox. I think we can delay the win screen. Also, it would be cool to make the camera face in front of the cat (instead of behind), hide the endbox, and play the win animation, and then show the win screen. I'm imagining something similar to when you beat a level in sonic adventure DX: https://youtu.be/fzGn-OEW2oM?t=1891
> 
> 1 -> It is playing, it's just very small since the side movement is just a tiny jump that is not too visible. I tried making it also move up when this animation is played so it's more noticeable but it looks weird and disturbs the other animations. Maybe we could brainstorm new side movement animations that are more noticeable on Saturday with the art team? 2 -> Yeaa there's a blink effect 3 -> Oh yeah I can look into that, that's a cool idea, but since the cat goes inside the box we would not really see their face. I could try having the cat stop before we get to the box and then continue moving forward, I don't think any of my other tasks have more priority than this one besides the playtesting report.

1. Hmm ok. Yeah because it's really not noticeable :( Hopefully you work it out with the art
2. Oh ok
3. Yeah I think that'd be a good workaround! (im still excited about doing that idea though)

**Review by judy-n — COMMENTED** [review]
The fall animation is so funny I love it!! 

But yeah I can't really see the side animation, I read that it's a very small jump so ok whatever. And I really can't see the win animation, once you make a change for that I'm happy with this! 🙌🏼

**Review by jamilfaisal — APPROVED** [review]
Good work!

**Pull request merged**